src/filter/SampleMetricFilter.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import annotations
+import abc
+from typing import TYPE_CHECKING, Type
+from src.filter.Filter import Filter
+from src.core.Setupable import SetupMode
+import pandas as pd
+import numpy as np
+from PIL import Image
+from matplotlib import pyplot as plt
+
+if TYPE_CHECKING:
+    from src.metric.SampleMetric import SampleMetric
+    from src.metric.SampleMetricManager import SampleMetricManager
+
+
+class SampleMetricFilter(Filter, metaclass=abc.ABCMeta):
+    """
+    Abstract static class outlining the general interface for different types
+    of sample metric filters.
+    """
+
+    def __new__(cls):
+        raise RuntimeError(f"{cls.__name__} should not be instantiated.")
+
+    @staticmethod
+    def sample_metric_reg_setup_modes(cls) -> dict[str, SetupMode]:
+        """
+        Returns setup modes for all relevent sample metrics.
+
+        # ! Note that if setup modes for sample metrics have the same name
+        # then only one of them (the last added) will be kept.
+
+        Returns:
+            dict[str, SetupMode]: The setup modes.
+        """
+        setup_modes = {}
+        # Merge all setup modes
+        for sm in cls.get_used_sample_metrics():
+            setup_modes |= sm.reg_setup_modes()
```

Comment:
cool operator! did not know that it existed

Reply:
New with python 3.9